sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -127,13 +128,19 @@ trait InvokeLike extends Expression with NonSQLExpression {
       arguments: Seq[Expression],
       input: InternalRow,
       dataType: DataType): Any = {
-    val args = arguments.map(e => e.eval(input).asInstanceOf[Object])
-    if (needNullCheck && args.exists(_ == null)) {
+    var i = 0
+    val len = arguments.length
+    while (i < len) {
+      val e = arguments(i)
```

Comment:
It looks we don't need this intermediate val?
```
evaluatedArgs(i) = arguments(i).eval(input).asInstanceOf[Object]
```

Reply:
yea let me remove it